app/workers/order_processor.rb

```diff
@@ -0,0 +1,43 @@
+class OrderProcessor
+  include Sidekiq::Worker
+  sidekiq_options backtrace: true, queue: :deliveries
+  def perform(consumed_order)
+    begin
+      puts 'sidekiq worker processed ' + consumed_order.to_s
+      consumed_order = JSON.parse(consumed_order)
+      order = Order.find_or_initialize_by(order_id: consumed_order['order_id'] )
+      if order.publish_time.nil? || order.publish_time < consumed_order['publish_time']
+        order.customer_name = consumed_order['customer_name']
+        order.time = consumed_order['time']
+        order.publish_time = consumed_order['publish_time']
+        order.status = consumed_order['status']
+      end
+      
+
+      extra_info = fill_extra_info consumed_order
+
+      if(order.info.nil?)
+        order.info = extra_info
+      else
+        order.info.merge(extra_info)
+      end
+
+      order.save!
+    rescue => exception
+      exception.to_s
+    end
+  end
+
+
+  private
+  def fill_extra_info (consumed_order)
+      extra_info = {}
+      extra_info[:courier] = consumed_order['courier'] if !consumed_order['courier'].nil?
```

Comment:
`!consumed_order['courier'].nil?` can be shortened to `consumed_order['courier'].present?` so it's not a double negative

Reply:
you could also shorten this to:
```ruby
extra_info_keys = [:courier, :cook, : estimated_time_of_arrival, :signed_by]
extra_info_keys.each do |key|
    extra_info[key] = consumed_order[key.to_s] if consumed_order[key.to_s].present?
end
```

Reply:
didn't use present as i understood that it is the opposite of "blank" which returns false if the value is a boolean true ... at least thats what was documented .... the loop is cool though 8-)